uaclient/cli.py

```diff
@@ -384,10 +403,13 @@ def _get_contract_token_from_cloud_identity(cfg: config.UAConfig) -> str:
                 ua_status.MESSAGE_UNSUPPORTED_AUTO_ATTACH
             )
         raise e
+    instance_id_file = os.path.join(cfg.data_dir, "instance-id")
+    if current_iid:
+        util.write_file(instance_id_file, current_iid)
```

Comment:
Should this be using `UAConfig.write_cache`?

Reply:
d'oh